src/v/kafka/server/tests/quota_manager_test.cc

```diff
@@ -223,3 +223,78 @@ SEASTAR_THREAD_TEST_CASE(static_config_test) {
         BOOST_CHECK_EQUAL(it->second->pm_rate->rate(), 1026);
     }
 }
+
+SEASTAR_THREAD_TEST_CASE(update_test) {
+    using clock = kafka::quota_manager::clock;
+    fixture f;
+    f.start().get();
+    auto stop = ss::defer([&] { f.stop().get(); });
+
+    set_config(basic_config).get();
+
+    auto now = clock::now();
+    {
+        // Update fetch config
+        ss::sstring client_id = "franz-go";
+        f.sqm.local().record_fetch_tp(client_id, 8194, now).get();
+        f.sqm.local()
+          .record_produce_tp_and_throttle(client_id, 8192, now)
+          .get();
+
+        set_config([](config::configuration& conf) {
+            auto fetch_config = YAML::Load(std::string(raw_basic_fetch_config));
+            for (auto n : fetch_config) {
+                n["quota"] = n["quota"].as<uint32_t>() + 1;
+            }
+            conf.kafka_client_group_fetch_byte_rate_quota.set_value(
+              fetch_config);
+        }).get();
+
+        // Check the rate has been updated
+        auto it = f.buckets_map.local()->find(client_id + "-group");
+        BOOST_REQUIRE(it != f.buckets_map.local()->end());
+        BOOST_REQUIRE(it->second->tp_fetch_rate.has_value());
+        BOOST_CHECK_EQUAL(it->second->tp_fetch_rate->rate(), 4098);
```

Comment:
suggestion: iiuc (doubtful), the original fetch bucket has been blown away at this point, so would it make sense to test that the throttle returns "no delay"?

Reply:
> suggestion: iiuc (doubtful), the original fetch bucket has been blown away at this point, so would it make sense to test that the throttle returns "no delay"? 

Yes, that's the current implementation, but a "better" implementation would carry across the existing delay, so I don't want to over-test.

Reply:
yeah fair point. I think there's some value in having a test that blows up when the implementation changes, as a reminder, but it might be overkill in this case.